### ISSUE TITLE
Continuous benchmark

### ISSUE THREAD

**Issue by EwenQuim** [go-fuego/fuego#285]
**Is your feature request related to a problem? Please describe.**

To avoid performance regressions.

**Describe the solution you'd like**

Something like this: https://gofiber.github.io/fiber/benchmarks/

**Describe alternatives you've considered**

**Additional context**

**Comment by Jeremie-Chauvel** [comment]
Hey!
Might be interested in looking into it over holidays. Will keep you posted!

**Comment by EwenQuim** [comment]
Nice! 

I spoke with Adrien Cacciaguerra, he was interested, but Codspeed does not support Go yet...

**Comment by Jeremie-Chauvel** [comment]
It was ofc my idea 😁

**Comment by ekumamatthew** [comment]
Could I be assigned to this?

**Comment by EwenQuim** [comment]
I thought @Jeremie-Chauvel wanted to work on it. Ask him!

But you already have 2 first good issues to work on :)

**Comment by Jeremie-Chauvel** [comment]
Hey @ekumamatthew, you can start, I will try to add a codspeed integration but I need to add go support to codspeed first. Having some benchmarks to upgrade to codspeed ones will help me anyway. Worst case scenario, I don't do the code speed integration and you still have benchmarks for Fuego.

TLDR: go for it

**Comment by EwenQuim** [comment]
Using https://github.com/benchmark-action/github-action-benchmark or native go benchmark with [benchstat](https://pkg.go.dev/golang.org/x/perf/cmd/benchstat) can be a good idea. Way easier than implementing Go support to Codspeed...

**Comment by EwenQuim** [comment]
As @ekumamatthew is already working on other issues, I suggest that you can work on this one if you'd like!

**Comment by EwenQuim** [comment]
Maybe try to update/reuse the Makefile commands.

But here it's a special issue because it's CI, so no guidelines!

**Comment by aidenwong812** [comment]
Understand. Thanks.

**Comment by adriencaccia** [comment]
Support for go on CodSpeed is in the works at https://github.com/CodSpeedHQ/codspeed-go 😉
We would love for you to try it out once it is ready
